### PR TITLE
Bash Auto Completion

### LINKED ISSUE #712: Enable auto complete or bash completion of commands in space-cli
>If you want this feature to be implemented, give it a thumbs up reaction, so that we can determine which features are important to you.
>👍

### PR DESCRIPTION
**Pull request by KavishShah09**
Fixes #712

### CODE REVIEW

**Comment by KavishShah09** [comment]
Task Linked: https://app.clickup.com/t/5rjrka

**Comment by sharadregoti** [comment]
1) Add dynamic completion for the flag --project
2) dynamic completion for account id for set & delete command
3) deploy command takes 2 flags of the filename, add extension filter to these commands
4) Add static completion to log-level flag
5) Make the --project flag as required. https://github.com/spf13/cobra/blob/master/bash_completions.md#mark-flags-as-required
6) Try adding dynamic completion for each of get sub command. for e.g ->
space-cli get db-schema [tab][tab] should show dbalias another [tab][tab] should show table-names in that db-Alias.
7) You haven't added static completion for a remove database command.
8) Don't create out.sh file. print script directly on the terminal

**Comment by KavishShah09** [comment]
> * Add dynamic completion for the flag --project
> * dynamic completion for account id for set & delete command
> * deploy command takes 2 flags of the filename, add extension filter to these commands
> * Add static completion to log-level flag
> * Make the --project flag as required. https://github.com/spf13/cobra/blob/master/bash_completions.md#mark-flags-as-required
> * Try adding dynamic completion for each of get sub command. for e.g ->
>   space-cli get db-schema [tab][tab] should show dbalias another [tab][tab] should show table-names in that db-Alias.
> * You haven't added static completion for a remove database command.
> * Don't create out.sh file. print script directly on the terminal
> * Space-cli get has 2 subcommands for doint same operation with singlular & plural forms. I want you to remove the singular form of sub commands

Can not add static/dynamic completions to flags hence can not do points 1,4
Can not add dynamic completion to commands since Command struct does not contain ValidArgsFunc in current version that we use so can not do point 6
Already have done points 2,7 and 9
The way we add flags is using StringP function which internally adds a flag using the Flag struct without any annotations so can not do point 3
Regarding point 5 I've tried it using the way from the link provided but it doesn't work upon testing it
Will do point 8 when all changes are finalized, for now need it to source the file

**Comment by KavishShah09** [comment]
@sharadregoti

**Comment by sharadregoti** [comment]
1) add CLI completion for [ZSH](https://github.com/spf13/cobra/blob/master/zsh_completions.md), & [powershell](https://github.com/spf13/cobra/blob/master/powershell_completions.md).
Add a flag **--file** this is a file name or a path to a file. the script generated should be written into this file. create the file if it doesn't exist.
The completion command will look like this **space-cli completions bash/zsh/powershell --file /etc/bash.completion.d/space-cli**

2) For Accounts Command, we have prefix checking operation for all the 3 subcommands. but as we now have auto-complete event providing the full correct name displays the selection prompt on the terminal. I think we should remove the prefix operation @YourTechBud what do you say?

3) In the previous, review I said to make --project flag as required. Add a TODO Note saying make the --project flag as required for add & remove registry command. But **markflagrequired** func is not working as expected

4) For apply command, @YourTechBud should we write a custom completion for filtering .yaml & subdirectories?

5) For deploy command, add filters for 2 extensions 
image

6) For get command, Add a TODO Note for dynamic completion of subcommands. As ValidArgFunction is not accessible 
   e-g. space-cli get dB-configs [tab][tab] show list of DB alias by querying space-cloud

File: space-cli/cmd/modules/auth/commands.go [hunk 1]
```diff
@@ -8,15 +8,28 @@ import (
...
-		RunE: actionGetAuthProviders,
+		Use:     "auth-providers",
+		Aliases: []string{"auth-provider"},
+		RunE:    actionGetAuthProviders,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			project, check := utils.GetProjectID()
+			if !check {
+				_ = utils.LogError("Project not specified in flag", nil)
+				return nil, cobra.ShellCompDirectiveDefault
+			}
```
Comment (sharadregoti) [inline space-cli/cmd/modules/auth/commands.go:20]: use log debug, everywhere. user should not get this error when tab tab is pressed

File: space-cli/cmd/modules/database/commands.go [hunk 2]
```diff
@@ -29,37 +31,114 @@ func GenerateSubCommands() []*cobra.Command {
...
 	var getrules = &cobra.Command{
-		Use:  "db-rules",
-		RunE: actionGetDbRules,
+		Use:     "db-rules",
+		Aliases: []string{"db-rule"},
+		RunE:    actionGetDbRules,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			switch len(args) {
+			case 0:
+				fmt.Printf("case 0")
```
Comment (sharadregoti) [inline space-cli/cmd/modules/database/commands.go:41]: remove println statement everywhere

**Comment by YourTechBud** [comment]
Whats the status on this

**Comment by KavishShah09** [comment]
@prithvi081099 is handling this @YourTechBud 
@prithvi081099  you need to sign the CLA

**Comment by sharadregoti** [comment]
@prithvi081099 auto-completion for account command is not working, fix this ASAP. & also sign the CLA

**Pull request merged**